src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +586,32 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static bool ValidateArguments(string[] args, CommandLineApplication app)
+    {
+        // Get all valid options from the configuration files
+        var validArguments = app.GetOptions().SelectMany(o => new[] { o.LongName, o.ShortName }).ToArray();
+
+        var argumentsNamesProvided = GetArgumentNames(args).Select(a => a.ToString()).ToArray();
+
+        foreach (var argumentName in argumentsNamesProvided)
+        {
+            // Check if the argument provided is a valid option/argument
+            if (!validArguments.Contains(argumentName))
```

Comment:
`validArguments` should be `HashSet` for quick lookup. Right now your code is O(N^2), with `HashSet` it will be O(N)